src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -9011,10 +9011,10 @@ static bool isCandidateUnique(ref MethodSymbol? method, MethodSymbol candidate)
                 (parameters.Any(p => p.EffectiveScope != DeclarationScope.Unscoped) ?
                 parameters.SelectAsArray(p => p.EffectiveScope) :
                 default);
-            var parameterDefaultValues = parameters.Any(p => p.HasExplicitDefaultValue) ?
+            var parameterDefaultValues = syntax.IsFeatureEnabled(MessageID.IDS_FeatureLambdaOptionalParameters) && parameters.Any(p => p.HasExplicitDefaultValue) ?
```

Comment:
Generally, we have a rank order of preferences for different approaches (from most preferred to least preferred):
1. bind the same, but emit different diagnostics based on LangVer
2. bind differently based on LangVer
3. parse differently based on LangVer

Was there an explicit design or compat discussion to go with (2) instead of (1)?

Reply:
I understand, but I'm wondering whether we should just take the break unconditionally. The inferred delegate type for `var` is only a year old. I think Chuck started a thread with Jared/Fred to discuss this explicitly.

Reply:
I have reverted the conditional checks and updated the doc.